### PR TITLE
fix errors when generic nodes fail to add slots

### PR DESCRIPTION
**Pull request by carlitosan**
Closes https://github.com/o3de/o3de/issues/7502

image

Signed-off-by: carlitosan <82187351+carlitosan@users.noreply.github.com>

### CODE REVIEW

**Review by lumberyard-employee-dm — APPROVED** [review]
I think we have reached the point where should try to convert the generic nodes to be functions reflected on the Behavior Context again.

That would be able to reduce the size of ScriptCanvas libraries under a GiB if were to do that.

**Comment by carlitosan** [comment]
> 

We're going to do this, but also slim down the BC burden by doing a lot of what BC does with code gen and the new translation system.

**Review by lsemp3d — APPROVED** [review]

File: Gems/ScriptCanvas/Code/Include/ScriptCanvas/Core/Node.h [hunk 1]
```diff
@@ -1098,7 +1098,7 @@ namespace ScriptCanvas
                 slotConfiguration.SetType(Data::FromAZType<AZStd::decay_t<ResultType>>());
                 slotConfiguration.SetConnectionType(ConnectionType::Output);
 
-                node.AddSlot(slotConfiguration);
+                AZ_VerifyError("ScriptCanvas", node.AddSlot(slotConfiguration).IsValid(), "Node failed to add a required Data Out slot");
```
Comment (AMZN-Gene) [inline Gems/ScriptCanvas/Code/Include/ScriptCanvas/Core/Node.h:1101]: Is there anything to add to the error message for how someone might go about debugging or fixing the error?

File: Gems/ScriptCanvas/Code/Include/ScriptCanvas/Core/Node.h [hunk 2]
```diff
@@ -1115,12 +1115,11 @@ namespace ScriptCanvas
...
             {
                 DataSlotConfiguration slotConfiguration;
-
                 slotConfiguration.m_name = t_Traits::GetResultName(Index);
                 slotConfiguration.SetType(Data::FromAZType<AZStd::decay_t<AZStd::tuple_element_t<Index, ResultType>>>());
-
                 slotConfiguration.SetConnectionType(connectionType);                
-                node.AddSlot(slotConfiguration);
+
+                AZ_VerifyError("ScriptCanvas", node.AddSlot(slotConfiguration).IsValid(), "Node failed to add a required Data Out slot");
```
Comment (AMZN-Gene) [inline Gems/ScriptCanvas/Code/Include/ScriptCanvas/Core/Node.h:1122]: ditto

**Review by AMZN-Gene — APPROVED** [review]

**Pull request merged**